### PR TITLE
Upgrade tokio, futures, hyper to released versions.

### LINKED ISSUE #117: Update to Tokio 0.2
As of now, hit `Poll::Pending` because the most recent `hyper`, `0.13.0-alpha.4`, requires `= 0.2.0-alpha.6`.

### PR DESCRIPTION
**Pull request by hdevalence**
Closes #117

### CODE REVIEW

File: zebra-network/src/protocol/external/codec.rs [hunk 1]
```diff
@@ -649,7 +650,7 @@ mod tests {
         let services = PeerServices::NODE_NETWORK;
         let timestamp = Utc.timestamp(1568000000, 0);
 
-        let rt = Runtime::new().unwrap();
+        let mut rt = Runtime::new().unwrap();
```
Comment (hdevalence) [inline zebra-network/src/protocol/external/codec.rs:653]: The tokio runtime doesn't use interior mutability any more so it has to be mutable to spawn tasks.

File: zebra-network/Cargo.toml [hunk 2]
```diff
@@ -21,13 +21,14 @@ rand = "0.7"
...
+futures = "0.3"
 
 tracing = "0.1"
-tracing-futures = { version = "0.1", features = ["tokio-alpha"], default-features = false }
+tracing-futures = "0.2"
 
-tower = "=0.3.0-alpha.2"
-tower-load = { git = "https://github.com/tower-rs/tower", branch = "v0.3.x"}
+tower = { git = "https://github.com/tower-rs/tower" }
+tower-load = { git = "https://github.com/tower-rs/tower" }
```
Comment (hdevalence) [inline zebra-network/Cargo.toml:32]: Better to just take the git version of tower until 0.3.0 is finished.

File: zebra-network/src/peer/handshake.rs [hunk 3]
```diff
@@ -1,14 +1,16 @@
...
     sync::{Arc, Mutex},
     task::{Context, Poll},
 };
 
 use chrono::Utc;
-use futures::channel::mpsc;
-use tokio::{codec::Framed, net::TcpStream, prelude::*, timer::Interval};
+use futures::{channel::mpsc, prelude::*};
+use tokio::net::TcpStream;
+use tokio_util::codec::Framed;
```
Comment (hdevalence) [inline zebra-network/src/peer/handshake.rs:13]: Tokio codec functionality has moved to `tokio_util`.

**Review by dconnolly — APPROVED** [review]
LGTM!

**Pull request merged**